js/kontent.js

```diff
@@ -0,0 +1,63 @@
+const projectId = "99b0b3b0-4838-0051-3d57-8af72f55e8a0";
+const apiBlogItems = `https://deliver.kontent.ai/${projectId}/items`;
+const apiBlogItem = `https://deliver.kontent.ai/${projectId}/items/`;
+const apiTaxonomies = `https://deliver.kontent.ai/${projectId}/taxonomies/`;
+const pageSize = 10;
+
+class URLBuilder {
+    constructor(baseURL) {
+        this.singleEndpoint = null;
+        this.includeTotalCount = true;
+        this.limit = null;
+        this.orderByElementProp = null;
+        this.blogTagsContains = null;
+        this.baseURL = new URL(baseURL + (this.singleEndpoint || ""));
+    }
+
+    build() {
+        const working = new URL(this.baseURL + (this.singleEndpoint || ""));
+        const searchParams = working.searchParams;
+        if (this.includeTotalCount) searchParams.set("includeTotalCount", this.includeTotalCount);
+        if (this.limit) searchParams.set("limit", this.limit);
+        if (this.orderByElementProp) searchParams.set("order", `elements.${this.orderByElementProp}`);
+        if (this.blogTagsContains) searchParams.set("elements.blog_tags[contains]", this.blogTagsContains);
+        return working;
+    }
+}
+const itemsFetchKontentApi = new URLBuilder(apiBlogItems);
+const itemFetchKontentApi = new URLBuilder(apiBlogItem);
+const taxonomiesFetchKontentApi = new URLBuilder(apiTaxonomies);
+
+
+export function buildBlogItemsUrl(tag) {
+    itemsFetchKontentApi.includeTotalCount = true;
+    itemsFetchKontentApi.limit = pageSize;
+    itemsFetchKontentApi.orderByElementProp = "post_date[desc]";
+    if (tag) itemsFetchKontentApi.blogTagsContains = tag;
+    return itemsFetchKontentApi.build();
+}
+
+
+export function buildBlogItemUrl(itemCodeName) {
+    itemFetchKontentApi.singleEndpoint = itemCodeName;
+    return itemFetchKontentApi.build();
+}
+
+
+export function buildTaxonomiesUrl() {
+    taxonomiesFetchKontentApi.singleEndpoint = "blog_tags";
+    return taxonomiesFetchKontentApi.build();
+}
+
+
+export async function fetchKontent(api) {
+    var resError = { items: [], error: true };
+    try {
+        const res = await fetch(api);
+        const json = await res.json();
```

Comment:
you might also consider adding the following check immediately after line 56:
```javascript
if (!res.ok) throw new Error(`${res.statusText} (${res.status})`);
```

(see https://developer.mozilla.org/en-US/docs/Web/API/Response#Properties)

Reply:
OK